builder/frameworks/espidf.py

```
@@ -102,25 +104,24 @@ def create_silent_action(action_func):
+    ARDUINO_FRAMEWORK_DIR = str(ARDUINO_FRAMEWORK_DIR_PATH)
     # Possible package names in 'package@version' format is not compatible with CMake
     if "@" in os.path.basename(ARDUINO_FRAMEWORK_DIR):
-        new_path = os.path.join(
-            os.path.dirname(ARDUINO_FRAMEWORK_DIR),
-            os.path.basename(ARDUINO_FRAMEWORK_DIR).replace("@", "-"),
-        )
+        new_path = str(Path(os.path.dirname(ARDUINO_FRAMEWORK_DIR)) / 
+                      os.path.basename(ARDUINO_FRAMEWORK_DIR).replace("@", "-"))
         os.rename(ARDUINO_FRAMEWORK_DIR, new_path)
         ARDUINO_FRAMEWORK_DIR = new_path
     assert ARDUINO_FRAMEWORK_DIR and os.path.isdir(ARDUINO_FRAMEWORK_DIR)
```

Ruff (0.12.2)

Use of `assert` detected

(S101)